### PR TITLE
HIVE-27293 Vectorization: Incorrect results with nvl for ORC table

### PR DESCRIPTION
**Pull request by ramesh0201**

### CODE REVIEW

**Review by rkirtir — APPROVED** [review]

**Review by difin — APPROVED** [review]
LGTM

File: storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/Decimal64ColumnVector.java [hunk 1]
```diff
@@ -136,12 +136,12 @@ public void setElement(int outputElementNum, int inputElementNum, ColumnVector i
           decimal64ColVector.vector[inputElementNum], decimal64ColVector.scale);
       scratchHiveDecWritable.mutateEnforcePrecisionScale(precision, scale);
       if (scratchHiveDecWritable.isSet()) {
-        vector[inputElementNum] = scratchHiveDecWritable.serialize64(scale);
+        vector[outputElementNum] = scratchHiveDecWritable.serialize64(scale);
```
Comment (aturoczy) [inline storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/Decimal64ColumnVector.java:139]: Wong indexer? It was not covered with unit test?

Comment (ramesh0201) [reply storage-api/src/java/org/apache/hadoop/hive/ql/exec/vector/Decimal64ColumnVector.java:139]: Yes, not for the primitive column vectors.

**Review by ayushtkn — APPROVED** [review]
LGTM

**Pull request merged**